### PR TITLE
OPTM-24 New Development Section

### PR DESCRIPTION
**Pull request by firstdayofjune**
- described the workflow, CI pipeline and directory structure
- added tips & tricks (flutter format file watcher)

OPTM-24 #close

### CODE REVIEW

**Comment by guisedlythe** [comment]
I propose some minor changes/fixes:

````diff
diff --git a/README.md b/README.md
index 90261fc..dd25440 100644
--- a/README.md
+++ b/README.md
@@ -1,12 +1,12 @@
 # P2P Task Manager
 
-A privacy-aware task manager. No account needed, all your tasks get synced using the local network only. 
+A privacy-aware task manager. No account needed, all your tasks get synced using the local network only.
 
 ## Getting Started
 
-Follow the instruction on https://flutter.dev/docs/get-started/install
+Follow the instruction on <https://flutter.dev/docs/get-started/install>
 
-Follow the instructions on https://flutter.dev/docs/get-started/editor
+Follow the instructions on <https://flutter.dev/docs/get-started/editor>
 
 Run `flutter pub get` in the terminal to load all dependencies
 
@@ -22,14 +22,14 @@ A few resources to get you started:
 - [Cookbook: Useful Flutter samples](https://flutter.dev/docs/cookbook)
 - [Online Documentation](https://flutter.dev/docs)
 
-# Development
+## Development
 
 Read this section, if you want to contribute to the project.
 
 ## Workflow
 
 1. Select a JIRA issue to work on.
-2. Create a feature branch for the issue, named _OPTM-XY_Meaninful-Branch-Name_)
+2. Create a feature branch for the issue, named _OPTM-XY_Meaningful-Branch-Name_)
 3. Write your code and add tests for it
 4. Format your files (`flutter run format <filenames>` or `dart run format <filenames>`)
 5. Run all tests (`flutter test`) and make sure they complete without errors or warnings
@@ -44,6 +44,7 @@ Every file gets checked for proper formatting. Also, all tests get executed.
 You can find all the commands run on CI in `.github/workflows/main.yaml`.
 
 ## Directory Structure
+
 ```
 root
 // Contains the application code
@@ -82,6 +83,6 @@ In Visual Studio Code:
 > Install the Flutter extension (see [Editor setup](https://flutter.dev/docs/get-started/editor)) to get automatic formatting of code in VS Code.
 >
 > To automatically format the code in the current source code window, right-click in the code window and
-select Format Document. You can add a keyboard shortcut to this VS Code Preferences.
+> select Format Document. You can add a keyboard shortcut to this VS Code Preferences.
 >
 > To automatically format code whenever you save a file, set the _editor.formatOnSave_ setting to true.
````

**Review by JacobRoe — CHANGES_REQUESTED** [review]
Beside the proposed changes by Konstantin, you might, in case you incorporate Konstantin's suggestions, want to adjust the child headers for the `Development` sections too.

I'm also not quite sure why we should use less then and greater than symbols for denoting a link. Typically, it gets rendered in blue anyway. Opinions? (Referring to Konstantin suggested changes.)

**Comment by guisedlythe** [comment]
> I'm also not quite sure why we should use less then and greater than symbols for denoting a link. Typically, it gets rendered in blue anyway. Opinions? (Referring to Konstantin suggested changes.)

https://github.com/DavidAnson/markdownlint/blob/v0.23.1/doc/Rules.md#md034

**Comment by JacobRoe** [comment]
> > I'm also not quite sure why we should use less then and greater than symbols for denoting a link. Typically, it gets rendered in blue anyway. Opinions? (Referring to Konstantin suggested changes.)
> 
> https://github.com/DavidAnson/markdownlint/blob/v0.23.1/doc/Rules.md#md034

Thank you. It also states the rationale at the end: "Without angle brackets, the URL isn't converted into a link by many markdown parsers." To be honest, it never happened to me that it didn't get correctly rendered (and I use markdown extensively). But at least now I see why it can make sense to add them. So, no objections anymore from my side.

**Comment by firstdayofjune** [comment]
I included the proposed changes and updated the README to reflect the new metrics in CI.

File: README.md [hunk 1]
```diff
@@ -1,49 +1,57 @@
...
 
-## CI Pipeline
+### CI Pipeline
 
 The Continuous Integration (CI) Pipeline runs whenever you push your changes to the repository.
-Every file gets checked for proper formatting. Also, all tests get executed.
+With each Pull Request the following happens:
 
-You can find all the commands run on CI in `.github/workflows/main.yaml`.
+- Files are formatted using `flutter format`
```
Comment (guisedlythe) [inline README.md]: Maybe a bit misleading: the CI checks if the files you committed are properly formatted, it doesn't actually format the files files for you.

File: README.md [hunk 2]
```diff
@@ -1,49 +1,57 @@
...
 
-## Getting Started
+## Setup
 
-Follow the instruction on https://flutter.dev/docs/get-started/install
+Follow the instruction on <https://flutter.dev/docs/get-started/install> to install flutter.
 
-Follow the instructions on https://flutter.dev/docs/get-started/editor
+Clone this repository and open a terminal window in the root directory:
+- Run `git submodule init && git submodule update` (this project uses the lww_crdt library as a submodule)
```
Comment (guisedlythe) [inline README.md]: Insert empty line before list: https://github.com/DavidAnson/markdownlint/blob/v0.23.1/doc/Rules.md#md032

File: README.md [hunk 3]
```diff
@@ -1,49 +1,57 @@
...
+- Files are formatted using `flutter format`
+- Tests are executed and checked for errors
+- Metrics are created (warnings added to the respective lines)
+- Codecov report is created (used as an indicator on code quality)
 
-## Directory Structure
+Find all the commands run on CI in `.github/workflows/main.yaml`.
+
+### Directory Structure
 ```
```
Comment (guisedlythe) [inline README.md]: Insert empty line after heading: https://github.com/DavidAnson/markdownlint/blob/v0.23.1/doc/Rules.md#md022

File: README.md [hunk 4]
```diff
@@ -72,16 +80,28 @@ You can create a file watcher to automatically format your files on save.
...
-select Format Document. You can add a keyboard shortcut to this VS Code Preferences.
+> select Format Document. You can add a keyboard shortcut to this VS Code Preferences.
 >
 > To automatically format code whenever you save a file, set the _editor.formatOnSave_ setting to true.
+
+## Run code metrics locally
+
+Run the following command to get information about the code metrics
+
+    flutter pub run dart_code_metrics:metrics lib
```
Comment (guisedlythe) [inline README.md]: Use consistent block style: https://github.com/DavidAnson/markdownlint/blob/v0.23.1/doc/Rules.md#md046
E.g.:
````
```sh
flutter pub run dart_code_metrics:metrics lib
```
````

File: README.md [hunk 5]
```diff
@@ -1,49 +1,57 @@
...
 
 1. Select a JIRA issue to work on.
-2. Create a feature branch for the issue, named _OPTM-XY_Meaninful-Branch-Name_)
+2. Create a feature branch for the issue, named _OPTM-XY_Meaningful-Branch-Name_
 3. Write your code and add tests for it
 4. Format your files (`flutter run format <filenames>` or `dart run format <filenames>`)
 5. Run all tests (`flutter test`) and make sure they complete without errors or warnings
-6. Write a descriptive commit message and commit your changes (use the [JIRA Smart Commit syntax](https://support.atlassian.com/jira-software-cloud/docs/process-issues-with-smart-commits/) to reference the issue)
-7. Push your changes and create a pull request into main when you are done.
+6. Run code metrics (`flutter pub run dart_code_metrics:metrics lib`) and fix warnings related to your changes
```
Comment (guisedlythe) [inline README.md]: Maybe just add a reference to the 'Run code metrics locally section'.
E.g.:
```md
[Run code metrics](#run-code-metrics-locally)
```

File: README.md [hunk 6]
```diff
@@ -72,16 +80,28 @@ You can create a file watcher to automatically format your files on save.
...
 >
 > To automatically format the code in the current source code window, right-click in the code window and
-select Format Document. You can add a keyboard shortcut to this VS Code Preferences.
+> select Format Document. You can add a keyboard shortcut to this VS Code Preferences.
 >
 > To automatically format code whenever you save a file, set the _editor.formatOnSave_ setting to true.
+
+## Run code metrics locally
+
+Run the following command to get information about the code metrics
```
Comment (guisedlythe) [inline README.md]: Maybe add that the following command runs the metrics only in the `lib` folder, use `flutter pub run dart_code_metrics:metrics lib test` to run it in the `test` folder as well.

**Review by guisedlythe — CHANGES_REQUESTED** [review]

**Review by guisedlythe — APPROVED** [review]
Approved, except the minor changes that I already pushed to this branch.

**Review by JacobRoe — APPROVED** [review]

**Pull request merged**